src/cmds/pbsdsh.c

```diff
@@ -95,6 +107,96 @@ bailout(int sig)
  * @return - Void
  *
  */
+
+
+int
+host_match(char *line, char *host)
+{
+	if (NULL == host ) return 0;
+	int len = strlen(line);
+	static char domain[PBS_MAXHOSTNAME + 1];
+	char fullhost[PBS_MAXHOSTNAME + 1];
+	static struct in_addr addr;
+	static int addrvalid = -1;
+
+	if (line[len - 1] == '\n')
+		line[len - 1] = '\0';
+
+	if (strcmp(line, host) == 0)
+		return 1;
+
+	if (addrvalid == -1) {
+		addr.s_addr = inet_addr(host);
+		addrvalid = (addr.s_addr == INADDR_NONE) ? 0 : 1;
+	}
+
+	if (addrvalid) { /* compare IP addresses */
+		struct hostent *hp = gethostbyname(line);
```

Comment:
According to my CentOS machine's man page for gethostbyname():
       The gethostbyname*(), gethostbyaddr*(), herror(), and hstrerror() func‐
       tions  are  obsolete.  Applications should use getaddrinfo(3), getname‐
       info(3), and gai_strerror(3) instead.


Maybe use the updated version?

Reply:
Yeah, POSIX standards suggest these no more, I could update it to  getaddrinfo version here.  However, as i said previously, the check part code for hostname is from pbs_tmrsh, as a reminder,  the code in pbs_tmrsh is left unchanged.

Reply:
Thanks for making the update.  pbs_tmrsh is quite an old command.  I can see why it is using the old calls.  Thanks for using the new ones.